CHANGELOG.md

```diff
@@ -31,6 +31,11 @@
   [Marcelo Fabri](https://github.com/marcelofabri)
   [#2628](https://github.com/realm/SwiftLint/issues/2628)
 
+* `discarded_notification_center_observer` rule now checks if the observer is
+  appended to an array before triggering the violation.
```

Comment:
Please add two spaces after this line. (And don't ask me why, it's just the style here, probably becaus of Markdown restrictions on GitHub).